packages/web/components/control/token-select-limit.tsx

```diff
@@ -0,0 +1,227 @@
+import { CoinPretty, PricePretty } from "@keplr-wallet/unit";
+import { DEFAULT_VS_CURRENCY } from "@osmosis-labs/server";
+import classNames from "classnames";
+import { observer } from "mobx-react-lite";
+import Image from "next/image";
+import { useRouter } from "next/router";
+import { FunctionComponent, useMemo, useState } from "react";
+
+import { TokenSelectDrawerLimit } from "~/components/drawers/token-select-drawer-limit";
+import { Disableable } from "~/components/types";
+import { EventName } from "~/config";
+import { useAmplitudeAnalytics, useWindowSize } from "~/hooks";
+import { OrderDirection } from "~/hooks/limit-orders";
+import { useCoinPrice } from "~/hooks/queries/assets/use-coin-price";
+import { useSwapAsset } from "~/hooks/use-swap";
+import { formatPretty } from "~/utils/formatter";
+
+export interface TokenSelectLimitProps {
+  dropdownOpen?: boolean;
+  setDropdownOpen?: (value: boolean) => void;
+  // TODO: Better typing
+  selectableAssets: ReturnType<typeof useSwapAsset>["asset"][];
+  baseAsset: ReturnType<typeof useSwapAsset>["asset"] &
+    Partial<{
+      amount: CoinPretty;
+      usdValue: PricePretty;
+    }>;
+  quoteAsset: ReturnType<typeof useSwapAsset>["asset"] &
+    Partial<{
+      amount: CoinPretty;
+      usdValue: PricePretty;
+    }>;
+  baseBalance: CoinPretty;
+  quoteBalance: CoinPretty;
+  onTokenSelect: (tokenDenom: string) => void;
+  canSelectTokens?: boolean;
+  orderDirection: OrderDirection;
+}
+
+export const TokenSelectLimit: FunctionComponent<
+  TokenSelectLimitProps & Disableable
+> = observer(
+  ({
+    dropdownOpen,
+    setDropdownOpen,
+    selectableAssets,
+    onTokenSelect,
+    canSelectTokens = true,
+    baseAsset,
+    quoteAsset,
+    baseBalance,
+    quoteBalance,
+    disabled,
+    orderDirection,
+  }) => {
+    const { isMobile } = useWindowSize();
+    const router = useRouter();
+    const { logEvent } = useAmplitudeAnalytics();
+
+    // parent overrideable state
+    const [isSelectOpenLocal, setIsSelectOpenLocal] = useState(false);
```

Comment:
BTW we have a `useControllableState` hook that may assist with this

Reply:
https://github.com/osmosis-labs/osmosis-frontend/pull/3308/commits/0c264e9f2c02e4d95002cbd82877722aca38a223